tests/test_functional.py

```diff
@@ -125,6 +125,51 @@ def test_simple(live_server, tmpdir, func_params):
     assert str(actual) == str(expected), expected
 
 
+@pytest.mark.parametrize(
+    'dataset_name', ['jpReview_books_reg.csv', 'jpReview_books_reg.csv.gz'])
+def test_simple_with_unicode(live_server, tmpdir, func_params, dataset_name):
+    # train one model in project
+    out = tmpdir.join('out.csv')
+    ui = PickableMock()
+    base_url = '{webhost}/predApi/v1.0/'.format(webhost=live_server.url())
+    ret = run_batch_predictions(
+        base_url=base_url,
+        base_headers={},
+        user='username',
+        pwd='password',
+        api_token=None,
+        create_api_token=False,
+        deployment_id=func_params['deployment_id'],
+        pid=func_params['pid'],
+        lid=func_params['lid'],
+        import_id=None,
+        n_retry=3,
+        concurrent=1,
+        resume=False,
+        n_samples=10,
+        out_file=str(out),
+        keep_cols=None,
+        delimiter=None,
+        dataset='tests/fixtures/{}'.format(dataset_name),
+        pred_name=None,
+        pred_threshold_name=None,
+        pred_decision_name=None,
+        timeout=None,
+        ui=ui,
+        auto_sample=False,
+        fast_mode=False,
+        dry_run=False,
+        encoding='',
+        skip_dialect=False
+    )
+
+    assert ret is None
+    actual = out.read_text('utf-8')
+    with open('tests/fixtures/jpReview_books_reg_out.csv', 'rU') as f:
+        expected = f.read()
+    assert str(actual) == str(expected), expected
```

Comment:
How does `str` works on Python 2? I mean `.read_text` should return `unicode`, and casting a unicode to `str` should probably fail o_O I'd expect to see here `six.text_type` instead of `str`.

Reply:
doesn't matter in case of output. Here we have:
row_id,0.0,1.0
0,1.0,0.0
1,1.0,0.0
2,1.0,0.0

so it can be converted as ascii